### ISSUE TITLE
Improve search on code.dlang.org

### ISSUE THREAD

**Issue by pbackus** [dlang/project-ideas#115]
## Description

The search feature on code.dlang.org often fails to return relevant results for the user's query.

- [Packages whose names include the search term as a substring are not found.][1]
- [A search for the exact name of a package sometimes fails to find it.][2]

[1]: https://github.com/dlang/dub-registry/issues/93
[2]: https://github.com/dlang/dub-registry/issues/417

## What are rough milestones of this project?

Fixing the issues linked above, dlang/dub-registry#93 and dlang/dub-registry#417.

## How does this project help the D community?

Improving code.dlang.org's search would make it easier for D programmers to find existing packages to use and/or contribute to.

## Recommended skills

Web application development.

## Point of Contact

@s-ludwig, @Geod24, @WebFreak001

## References

Previous attempts:

- https://github.com/dlang/dub-registry/pull/481
- https://github.com/dlang/dub-registry/pull/497

**Comment by 0xEAB** [comment]
I’d just use an established search engine/database for this, e.g. meilisearch.
Sounds more maintainable than rolling your own.

**Comment by dejlek** [comment]
@pbackus check this PR and see if that is what you had in mind: https://github.com/dlang/dub-registry/pull/600

**Comment by pbackus** [comment]
@dejlek From the description, it sounds like it fixes dlang/dub-registry#93, but I don't understand the code well enough to tell if it solves dlang/dub-registry#417. Either way, it's a big improvement.

**Comment by elementgreen** [comment]
@pbackus Yes, it does resolve [dlang/dub-registry#93](https://github.com/dlang/dub-registry/issues/93).  In regards to [dlang/dub-registry#417](https://github.com/dlang/dub-registry/issues/417).  That has to do with the dash being interpreted as a word separator and it then searches with a logic OR which seems fairly useless for this application.  Indeed, **matplotlib-d** has the same results as **matplotlib d**.  Double quoting it **"matplotlib-d"** as mentioned in [#490](https://github.com/dlang/dub-registry/issues/490) in one of the comments does appear to search for the exact match.

It doesn't seem like allowing special characters in this search feature makes sense though, so I pushed an additional commit to my [PR](https://github.com/dlang/dub-registry/pull/600) which escapes special characters (backslashes and double-quotes) and then double quotes the string when doing the FTS search, so that any value entered is taken as the literal search term.

**Comment by WebFreak001** [comment]
I think this can be closed (no permissions)
